tests/lct_test.py

```diff
@@ -20,12 +20,97 @@
 ]
 
 
-def test_lct_signal():
+class _MultiCases:
+    def __init__(self):
+        self._values = self._vals()
```

Comment:
Does these need to be cached? If so, then _vals and _sigs nested functions to `__init__` and assign to self._sigs, _vals.